libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationTelemetry.java

```diff
@@ -177,18 +177,14 @@ void startSession(DirectionsRoute directionsRoute) {
   }
 
   void stopSession() {
-    if (navigationSessionState.startTimestamp() != null) {
-      NavigationMetricsWrapper.cancelEvent(
-        navigationSessionState, metricProgress, metricLocation.getLocation(), context
-      );
-    }
     gpsEventFactory.reset();
     resetDepartFactory();
   }
 
   void endSession(boolean isConfigurationChange) {
     this.isConfigurationChange = isConfigurationChange;
     if (!isConfigurationChange) {
+      sendCancelEvent();
```

Comment:
Rather than move the cancel event back into `endSession`, I think we should adjust where `endSession` itself is invoked (currently `Activity#onDestroy`).  Maybe it makes more sense to do it in `Service#onDestroy`?  As this should be guaranteed to occur after `stopNavigation` > `stopSession`

Reply:
Yeah that's another approach too which makes sense, will update.

Reply:
This is updated @danesfeder 

> Approach finally taken is to keep sending the cancel event within `NavigationTelemetry#stopSession` but move the `NavigationTelemetry#endSession` call from `NavigationLifecycleMonitor#onActivityDestroyed` to `NavigationService#endNavigation`. We should test thoroughly that this change doesn't introduce any other issues. 

Let me know what you think.